components/keyrings/keyring_kms/backend/backend.h

```diff
@@ -0,0 +1,158 @@
+/* Copyright (c) 2021, Oracle and/or its affiliates.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License, version 2.0,
+   as published by the Free Software Foundation.
+
+   This program is also distributed with certain software (including
+   but not limited to OpenSSL) that is licensed under separate terms,
+   as designated in a particular file or component or in included license
+   documentation.  The authors of MySQL hereby grant you an additional
+   permission to link the program and your derivative works with the
+   separately licensed software that they have included with MySQL.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License, version 2.0, for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#ifndef KEYRING_KMS_BACKEND_INCLUDED
+#define KEYRING_KMS_BACKEND_INCLUDED
+
+#include <string>
+
+#include <components/keyrings/common/json_data/json_writer.h>
+#include <components/keyrings/common/memstore/iterator.h>
+#include <components/keyrings/common/operations/operations.h>
+
+#include "config/config.h"
+#include "kms.h"
+
+namespace keyring_kms {
+namespace backend {
+class Keyring_kms_backend final {
+ public:
+  explicit Keyring_kms_backend(const std::string keyring_kms_name,
```

Comment:
```suggestion
  explicit Keyring_kms_backend(const std::string &keyring_kms_name,
```
`explicit` does not make much sense for constructors with more than 1 argument